pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -653,7 +653,7 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 	availableTopologyAligned := available
 
 	sockets := make(map[int]bool)
-	socketsArray := containerTopologyHint.SocketMask.GetSockets()
+	socketsArray := containerTopologyHint.SocketAffinity.GetSockets()
```

Comment:
As I said in my earlier comment, I think that containerTopologyHint can be nil if the feature gate is disabled.  I think we need a conditional around this whole section of code.